test/promisification_tests.js

```diff
@@ -3,103 +3,88 @@ const assert = require('assert');
+  }
+  catch(e) {
+    assert(e instanceof Error);
+  }
 });
 
-test("[promisified client] should send an event as udp", function(done) {
-  client.send(client.Event({
+test("[promisified client] should send an event as udp", async function() {
+  await client.send(client.Event({
     service : 'hello_udp',
     metric  : Math.random(100)*100,
-    tags    : ['bar'] }), client.udp).then(done);
+    tags    : ['bar'] }), client.udp);
 });
 
-test("[promisified client] should send an event as tcp", function(done) {
+test("[promisified client] should send an event as tcp", async function() {
   let counter = 10;
 
-  function listener(data) {
-    assert(data.ok);
-    if (--counter === 0) {
-      setTimeout(done, 100);
-    }
-  };
-
   for (let i = Number(counter); i >= 0; i--) {
-    client.send(client.Event({
+    const data = await client.send(client.Event({
       service : 'hello_tcp_'+i,
       metric  : Math.random(100)*100,
-      tags    : ['bar'] }), client.tcp).then(listener);
-  }
+      tags    : ['bar'] }), client.tcp);
 
+    assert(data.ok);
+  }
 });
 
-test("[promisified client] should send an event with custom attributes as tcp", function(done) {
+test("[promisified client] should send an event with custom attributes as tcp", async function() {
   let counter = 10;
 
-  function listener(data) {
-    assert(data.ok);
-    if (--counter === 0) {
-      setTimeout(done, 100);
-    }
-  };
-
   for (let i = Number(counter); i >= 0; i--) {
-    client.send(client.Event({
+    const data = await client.send(client.Event({
       service : 'hello_tcp_'+i,
       attributes: [{key: "session", value: "123-456-789"},
                    {key: "metric_type", value: "random_number"}
                    ],
       metric  : Math.random(100)*100,
-      tags    : ['bar'] }), client.tcp).then(listener);
+      tags    : ['bar'] }), client.tcp);
+
+    assert(data.ok);
   }
 });
 
-test("[promisified client] should send a state as udp", function(done) {
-  client.send(client.State({
+test("[promisified client] should send a state as udp", async function() {
+  await client.send(client.State({
     service: 'state_udp',
     state: 'ok'
-  }), client.udp).then(done);
+  }), client.udp);
 });
 
-test("[promisified client] should send a state as tcp", function(done) {
+test("[promisified client] should send a state as tcp", async function() {
   let counter = 10;
 
-  function listener(data) {
-    assert(data.ok);
-    if (--counter === 0) {
-      setTimeout(done, 100);
-    }
-  };
-
   for (let i = Number(counter); i >= 0; i--) {
-    client.send(client.State({
+    const data = await client.send(client.State({
       service: 'state_tcp',
       state: 'ok'
-    }), client.tcp).then(listener);
+    }), client.tcp);
+
+    assert(data.ok);
   }
 });
 
-test("[promisified client] should query the server", function(done) {
-  client.send(client.Query({
+test("[promisified client] should query the server", async function() {
```

Comment:
I haven't done node in a while, is async/await fully compatible with all existing releases? Does this change the support window of node versions for dependents if they upgrade this library?

Reply:
Async/Await in my understanding is just a syntactic sugar on the top of regular promises.
So if one is under Node 7.6, he/she can use promises with .then/.catch and starting from 7.6 - with async/await.

Reply:
Ok that makes sense but this sugar is being used within the library code itself, so wouldn't that require library users to upgrade the minimum node version to use it?

Reply:
As you can see, I do not use async/await in library code, only in tests.
Those who want to run the suite should update their Node to 7.6 or higher, if we decide to keep async/await style for the promisification tests.